GetDKAN/dkan
# Add Drush command to apply a dictionary to a resource

Fixes [issue#]

## TO DO

While testing we noticed that applying the data dictionary via the `dictionaryEnforcer `would apply the dictionary but would not update the status under the post import column of the dashboard (`/admin/dkan/datastore/status`). A better approach might be to utilize the `PostImport` service and call `processResource`. By using `PostImport` instead of the `dictionaryEnforcer` we are able capture the post import status.

## Describe your changes

Adds a drush command: `dkan:datastore:apply-dictionary`

This command applies the data dictionary to the specified dataset, as configured in the site, and as specified by `describedBy`.

## QA Steps

- Set the site to not use data dictionaries.
- Add a data dictionary to a dataset.
- Set the site to use the data dictionary.
- Run `drush dkan:datastore:apply-dictionary` for the dataset.
- Verify that the data dictionary is displayed for the dataset.
- If indexing was specified in the dictionary, verify that the indexing was applied to the database table for the resource.

## Checklist before requesting review

_If any of these are left unchecked, please provide an explanation_
